src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,489 @@
+                ECPoint[] oraclePublicKeys = NativeContract.RoleManagement.GetDesignatedByRole(snapshot, Role.Oracle, snapshot.Height + 1);
+                foreach (var account in wallet.GetAccounts())
+                {
+                    var oraclePub = account.GetKey()?.PublicKey;
+                    if (!account.HasKey || account.Lock || !oraclePublicKeys.Contains(oraclePub)) continue;
+
+                    var txSign = responseTx.Sign(account.GetKey());
+                    var backTxSign = backupTx.Sign(account.GetKey());
+                    AddResponseTxSign(snapshot, requestId, oraclePub, txSign, responseTx, backupTx, backTxSign);
+                    tasks.Add(SendResponseSignatureAsync(requestId, txSign, account.GetKey()));
+
+                    Log($"Send oracle sign data: Oracle node: {oraclePub} RequestTx: {request.OriginalTxid} Sign: {txSign.ToHexString()}");
+                }
+                await Task.WhenAll(tasks);
+            }
+        }
+
+        private async void ProcessRequestsAsync()
+        {
+            while (!cancelSource.IsCancellationRequested)
+            {
+                using (var snapshot = Blockchain.Singleton.GetSnapshot())
+                {
+                    foreach (var (id, request) in NativeContract.Oracle.GetRequests(snapshot))
+                    {
+                        if (cancelSource.IsCancellationRequested) break;
+                        if (!finishedCache.ContainsKey(id) && (!pendingQueue.TryGetValue(id, out OracleTask task) || task.Tx is null))
+                            await ProcessRequestAsync(snapshot, request);
+                    }
+                }
+                if (cancelSource.IsCancellationRequested) break;
+                await Task.Delay(500);
+            }
+            stopped = true;
+        }
+
+        private async Task<(OracleResponseCode, string)> ProcessUrlAsync(string url)
+        {
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var uri))
+                return (OracleResponseCode.Error, null);
+            if (!protocols.TryGetValue(uri.Scheme, out IOracleProtocol protocol))
+                return (OracleResponseCode.ProtocolNotSupported, null);
+            try
+            {
+                return await protocol.ProcessAsync(uri, cancelSource.Token);
+            }
+            catch
+            {
+                return (OracleResponseCode.Error, null);
+            }
+        }
+
+        public static Transaction CreateResponseTx(StoreView snapshot, OracleRequest request, OracleResponse response, ECPoint[] oracleNodes)
+        {
+            var requestTx = snapshot.Transactions.TryGet(request.OriginalTxid);
+            var n = oracleNodes.Length;
+            var m = n - (n - 1) / 3;
+            var oracleSignContract = Contract.CreateMultiSigContract(m, oracleNodes);
+
+            var tx = new Transaction()
+            {
+                Version = 0,
+                Nonce = 0,
```

Comment:
Use random number?

Reply:
Done

Reply:
We need a fixed value otherwise other oracle nodes won't sign the same transaction

Reply:
Please check https://github.com/neo-project/neo-modules/pull/326/commits/31c0d98804ac2f7ca275551ac1ed6fd17eaa6fc8

Reply:
Use the requestId?